StarVMC/StarAgmlLib/AgMLExtension.cxx

```diff
@@ -9,9 +10,31 @@ AgMLExtension::AgMLExtension() : TGeoRCExtension(),
 				 mBranchings(0),
 				 mVolumeId( new AgMLVolumeId ),
 				 mHitScoring(),
-				 mGstpar()
+				 mGstpar(),
+				 mEngine(-1)
 {
   Grab();
 }
 
+
 void AgMLExtension::AddCut( TString cut, double value ){ mGstpar[cut] = value; }
+
+void AgMLExtension::Print( Option_t* opts ) const {
+
+//static const char* en[] = { "default", "geant3", "geant4" };
+
+  LOG_INFO << mModuleName.Data() << ":"
+	   << mFamilyName.Data() << ":"
+	   << mVolumeName.Data() << " this="
+	   << this << " sens="
+	   << mSensitive << " tracking="
+	   << mTracking << " nbranch="
+	   << mBranchings << " nuser="
+    //	   << mHitScoring.size() << " engine="
+    //    	   << (mEngine>=0 && mEngine < 2) ? en[mEngine] : "invalid!? " 
+	   << endm;
+
+}
+
+std::map<TString, AgMLExtension*> AgMLExtension::mExtensionMap;
```

Comment:
The GEANT3 VMC took use of both the "user" and "framework" extensions to TGeoVolume, leaving us with the only option of keeping a map between volume names and extension data.